client/js/helpers/roundBadgeNumber.ts

```diff
@@ -3,5 +3,9 @@ export default (count: number) => {
 		return count.toString();
 	}
 
-	return (count / 1000).toFixed(2).slice(0, -1) + "k";
+	const suffixes = ["", "k", "M"];
+	const magnitudeIndex = Math.min(Math.floor(Math.log10(count) / 3), suffixes.length - 1);
```

Comment:
While I understand the use case here, I think it's simpler to just have two if cases to handle `M/k`.
```
if(count >= 1000000) M
if(count >= 1000) k
count
```